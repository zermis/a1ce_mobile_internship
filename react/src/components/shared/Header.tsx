import React from 'react';
import { Dimensions, StatusBar } from 'react-native';
import { StyleSheet, View } from 'react-native';
import { LinearGradient } from 'expo-linear-gradient';

import Colors from '../../constants/Colors';

interface Props {
    children?: JSX.Element;
    style?: object;
}

const { height } = Dimensions.get('window');

const Header = (props: Props): JSX.Element => {
    return (
        <View style={styles.headerContainer}>
            <LinearGradient
                colors={Colors.linearGradient}
                style={styles.linearGradient}
                start={{ x: 0, y: 0.5 }}
                end={{ x: 1, y: 0.5 }}
            >
                <StatusBar translucent={true} backgroundColor={'transparent'} />
                <View style={{ ...styles.headerContent, ...props.style }}>
                    {props.children}
                </View>
            </LinearGradient>
        </View>
    );
};

const styles = StyleSheet.create({
    headerContainer: {
        height: height > 600 ? '35%' : '46%',
        borderBottomLeftRadius: 30,
        borderBottomRightRadius: 30,
        overflow: 'hidden'
    },
    linearGradient: {
        flex: 1
    },
    headerContent: {
        flex: 1,
        marginLeft: height * 0.06,
        marginRight: height * 0.06,
        justifyContent: 'center'
    }
});

export default Header;
