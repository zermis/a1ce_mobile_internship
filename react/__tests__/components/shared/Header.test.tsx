import React from 'react';
import { View, Text } from 'react-native';
import renderer from 'react-test-renderer';
import Header from '../../../src/components/shared/Header';

it('renders correctly', () => {
    const tree = renderer
        .create(
            <Header>
                <View>
                    <Text>Hello World</Text>
                </View>
            </Header>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
