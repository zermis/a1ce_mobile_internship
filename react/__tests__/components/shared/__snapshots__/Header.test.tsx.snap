// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "borderBottomLeftRadius": 30,
      "borderBottomRightRadius": 30,
      "height": "35%",
      "overflow": "hidden",
    }
  }
>
  <ViewManagerAdapter_ExpoLinearGradient
    proxiedProperties={
      Object {
        "colors": Array [
          4289279286,
          4281878109,
        ],
        "endPoint": Array [
          1,
          0.5,
        ],
        "locations": undefined,
        "startPoint": Array [
          0,
          0.5,
        ],
      }
    }
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "justifyContent": "center",
          "marginLeft": 80.03999999999999,
          "marginRight": 80.03999999999999,
        }
      }
    >
      <View>
        <Text>
          Hello World
        </Text>
      </View>
    </View>
  </ViewManagerAdapter_ExpoLinearGradient>
</View>
`;
